### PR TITLE
Replace state income tax with estimated withholding for all states in income tax used for SALT deduction

### LINKED ISSUE #4024: Replace state income tax with estimated withholding for all states in income tax used for SALT deduction
Continuing from #4017 for consistency.

### PR DESCRIPTION
**Pull request by Shinuing**
Fixes #4024 

create basic structure

### CODE REVIEW

File: policyengine_us/variables/gov/states/in/tax/income/in_withheld_income_tax.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+    value_type = float
+    entity = Person
+    label = "Indiana withheld income tax"
+    defined_for = StateCode.IN
+    unit = USD
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        employment_income = person("irs_employment_income", period)
+        p = parameters(period).gov.states.ca.tax.income
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/in/tax/income/in_withheld_income_tax.py:14]:
```suggestion
        p = parameters(period).gov.states["in"]tax.income
```

File: policyengine_us/variables/gov/states/ks/tax/income/ks_withheld_income_tax.py [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    value_type = float
+    entity = Person
+    label = "Kansas withheld income tax"
+    defined_for = StateCode.KS
+    unit = USD
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        employment_income = person("irs_employment_income", period)
+        p = parameters(period).gov.states.ca.tax.income
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/ks/tax/income/ks_withheld_income_tax.py:14]:
```suggestion
        p = parameters(period).gov.states.ka.tax.income
```

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/ks/tax/income/ks_withheld_income_tax.py:14]: and others

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/parameters/gov/state_income_tax.yaml
Comment (PavelMakarchuk) [inline policyengine_us/parameters/gov/state_income_tax.yaml:1]: some states use the federal SD and some states use the personal exemptions instead

Comment (Shinuing) [reply policyengine_us/parameters/gov/state_income_tax.yaml:1]: yes, I'm not finishing yet. Just created basic structure for all states needed. will check when doing

**Comment by Shinuing** [comment]
Some confusions @PavelMakarchuk 

1. Do we always apply maximum sd compared with base sd?
2. **KS**: do i need to model JOINT separately? JOINT rate is different than others
2: **MI**: sd tier 2 or tier 3 ?
3. **PA**: Pennsylvania does not have standard deductions, personal exemptions, or itemized deductions. Also, They do not use the federal standard deduction amounts. How can we do withheld in this situation?

File: changelog_entry.yaml
Comment (PavelMakarchuk) [inline changelog_entry.yaml:1]: Some tests fail

File: policyengine_us/variables/gov/states/id/tax/income/id_withheld_income_tax.py [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    entity = Person
+    label = "Idaho withheld income tax"
+    defined_for = StateCode.ID
+    unit = USD
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        employment_income = person("irs_employment_income", period)
+        p_irs = parameters(period).gov.irs.deductions.standard
+        # We apply the base standard deduction amount
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/id/tax/income/id_withheld_income_tax.py:15]:
```suggestion
        # We apply the base IRS standard deduction amount
```

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/id/tax/income/id_withheld_income_tax.py:15]: all files with IRS

File: policyengine_us/tests/policy/baseline/gov/states/nm/tax/income/nm_withheld_income_tax.yaml [hunk 4]
```diff
@@ -8,7 +8,7 @@
   output:
     nm_withheld_income_tax: 34.85
 
-- name: Changing the filing sttaus should not change the output 
+- name: Changing the filing status should not change the output
```
Comment (PavelMakarchuk) [inline policyengine_us/tests/policy/baseline/gov/states/nm/tax/income/nm_withheld_income_tax.yaml:11]: thanks for that!

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/pa/tax/income/pa_withheld_income_tax.py [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+    value_type = float
+    entity = Person
+    label = "Pennsylvania withheld income tax"
+    defined_for = StateCode.PA
+    unit = USD
+    definition_period = YEAR
+
+    def formula(person, period, parameters):
+        employment_income = person("irs_employment_income", period)
+        p = parameters(period).gov.states.pa.tax.income
```
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/pa/tax/income/pa_withheld_income_tax.py:14]: finish var.

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

File: policyengine_us/variables/gov/states/pa/tax/income/pa_withheld_income_tax.py (hunk 5 above)
Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/pa/tax/income/pa_withheld_income_tax.py:14]: also add test

Comment (Shinuing) [reply policyengine_us/variables/gov/states/pa/tax/income/pa_withheld_income_tax.py:14]: Hi please check my comment above, I have mentioned my concerns with PA also other states

**Comment by PavelMakarchuk** [comment]
> Some confusions @PavelMakarchuk
> 
> 1. Do we always apply maximum sd compared with base sd?
> 2. **KS**: do i need to model JOINT separately? JOINT rate is different than others
> 3. **MI**: sd tier 2 or tier 3 ?
> 4. **PA**: Pennsylvania does not have standard deductions, personal exemptions, or itemized deductions. Also, They do not use the federal standard deduction amounts. How can we do withheld in this situation?

1. We apply the max SD if there is a phase-out but if there if thee is an additional SD (e.g. dependents) we use the base 
2. We only model SINGLE 
3. MI -> lets do no SD as they only apply for elderly (over 65) - same with exemptions
4. Lets just take the rate here

File: policyengine_us/variables/gov/states/nh/tax/income/nh_withheld_income_tax.py
Comment (PavelMakarchuk) [inline policyengine_us/variables/gov/states/nh/tax/income/nh_withheld_income_tax.py:1]: I dont think new hampshire taxes only dividend and interest - so there is no withheld tax - we should NH completely from the state income tax yaml file

Comment (PavelMakarchuk) [reply policyengine_us/variables/gov/states/nh/tax/income/nh_withheld_income_tax.py:1]: (the way e.g. Texas is)

**Review by PavelMakarchuk — CHANGES_REQUESTED** [review]

**Review by PavelMakarchuk — APPROVED** [review]

**Review by MaxGhenis — APPROVED** [review]

**Pull request merged**